httpx/models.py

```diff
@@ -613,7 +620,16 @@ def prepare(self) -> None:
         has_connection = "connection" in self.headers
 
         if not has_host:
-            auto_headers.append((b"host", self.url.authority.encode("ascii")))
+            url = self.url
+            if url.userinfo:
+                url = url.copy_with_urlparse(userinfo=None)
+                for userinfo in ("username", "password"):
```

Comment:
We don't want to add "username" and "password" headers to the request.
I think we can drop all of this "for" block.

Reply:
Ah sorry, I applied like stdlib
>>> o = urlparse('http://uname@www.cwi.nl:80/%7Eguido/Python.html')
>>> o.username
'uname'